test/integration/kubernetes_deploy_test.rb

```diff
@@ -672,6 +676,57 @@ def test_bad_container_on_daemon_sets_fails
     ], in_order: true)
   end
 
+  def test_bad_container_on_stateful_sets_fails_with_rolling_update
+    skip if KUBE_CLIENT_VERSION < Gem::Version.new("1.7.0")
+    result = deploy_fixtures("hello-cloud", subset: ["stateful_set.yml"]) do |fixtures|
+      stateful_set = fixtures['stateful_set.yml']['StatefulSet'].first
+      stateful_set['spec']['updateStrategy'] = {}
+      stateful_set['spec']['updateStrategy']['type'] = 'RollingUpdate'
+      container = stateful_set['spec']['template']['spec']['containers'].first
+      container["image"] = "busybox"
+      container["command"] = ["ls", "/not-a-dir"]
+    end
+
+    assert_deploy_failure(result)
+    assert_logs_match_all([
+      "StatefulSet/nginx-ss: FAILED",
+      "nginx: Crashing repeatedly (exit 1). See logs for more information.",
+      "Events (common success events excluded):",
+      "[Pod/nginx-ss-0]	FailedSync: Error syncing pod",
+      "Logs from container 'nginx' (last 250 lines shown):",
+      "ls: /not-a-dir: No such file or directory"
+    ], in_order: true)
+  end
+
+  def test_on_delete_stateful_sets_are_not_monitored
+    result = deploy_fixtures("hello-cloud", subset: ["stateful_set.yml"])
+
+    assert_deploy_success(result)
+    assert_logs_match_all([
+      "WARNING: Your StatefulSet's updateStrategy is set to OnDelete",
+      "Successful resources",
+      "StatefulSet/nginx-ss"
+    ], in_order: true)
+  end
+
+  def test_rolling_update_stateful_sets_are_monitored
+    skip if KUBE_CLIENT_VERSION < Gem::Version.new("1.7.0")
+    result = deploy_fixtures("hello-cloud", subset: ["stateful_set.yml"]) do |fixtures|
+      stateful_set = fixtures['stateful_set.yml']['StatefulSet'].first
+      stateful_set['spec']['updateStrategy'] = {}
+      stateful_set['spec']['updateStrategy']['type'] = 'RollingUpdate'
+    end
+
+    assert_deploy_success(result)
+    assert_logs_match_all([
+      "Deploying resources:",
+      "Successfully deployed",
+      "Service/nginx-ss",
+      "Successful resources",
+      "StatefulSet/nginx-ss"
```

Comment:
Let's assert on the `2 replicas, 2 readyReplicas, 2 currentReplicas` output here, since we'd normally do that in the "hello cloud" test but can't because it uses OnDelete.